iotedgehubdev/cli.py

```diff
@@ -300,12 +301,61 @@ def validateconfig():
     _parse_config_json()
     output.info('Config file is valid.')
 
+@click.command(context_settings=CONTEXT_SETTINGS,
+               help="Create IoT Edge device CA")
+@click.option('--output-dir',
+              '-o',
+              required=False,
+              default=".",
+              help='The output folder of generated certs. '
+              'The tool will create a certs folder under given path to store the certs.')
+@click.option('--valid-days',
+              '-d',
+              required=False,
+              default=90,
+              show_default=True,
+              help='Days before cert expires.')
+@click.option('--force',
+              '-f',
+              required=False,
+              is_flag=True,
+              default=False,
+              show_default=True,
+              help='Whether overwrite existing cert files.')
+@click.option('--trusted-ca',
+              '-c',
+              required=False,
+              help='Path of your own trusted ca used to sign IoT Edge device ca. '
+              'Please also provide trsuted ca private key and related passphase (if have).'
+              )
+@click.option('--trusted-ca-key',
+              '-k',
+              required=False,
+              help='Path of your own trusted ca private key used to sign IoT Edge device ca. '
+              'Please also provide trusted ca and related passphase (if have).')
+@click.option('--trusted-ca-key-passphase',
+              '-p',
+              required=False,
+              help='Passphase of your own trusted ca private key.')
+@_with_telemetry
+def generatedeviceca(output_dir, valid_days, force, trusted_ca, trusted_ca_key, trusted_ca_key_passphase):
+    try:
+        output_dir = os.path.abspath(os.path.join(output_dir, EdgeConstants.CERT_FOLDER))
+        if trusted_ca_key_passphase:
+            trusted_ca_key_passphase = trusted_ca_key_passphase.encode()  # crypto requires byte string
+        edgeCert = EdgeCert(output_dir, '')
+        edgeCert.generate_device_ca(valid_days, force, trusted_ca, trusted_ca_key, trusted_ca_key_passphase)
+        output.info('Successfully generated device ca. Please find the generated certs at %s' % output_dir)
```

Comment:
also show the "certs" folder in output?

Reply:
Already appended the "certs" folder to output_dir at the beginning of the function, so the path will contain the "certs" folder